tests/src/Integration/Condition/UserHasRoleTest.php

```diff
@@ -88,4 +88,28 @@ public function testConditionEvaluation() {
     $this->assertFalse($this->condition->evaluate());
   }
 
+  /**
+   * Test the behavior with unsupported operations.
+   *
+   * @covers ::execute
+   *
+   * @expectedException \InvalidArgumentException
+   */
+  public function testInvalidOperationException() {
+    // Set-up a mock object with roles 'authenticated' and 'editor', but not
+    // 'administrator'.
+    $account = $this->prophesizeEntity(UserInterface::class);
+    $account->getRoles()->willReturn(['authenticated', 'editor']);
+
+    $this->condition->setContextValue('user', $account->reveal());
+
+    $authenticated = $this->prophesize(RoleInterface::class);
+    $authenticated->id()->willReturn('authenticated');
+
+    // Now test INVALID as operation. An exception must be thrown.
+    $this->condition->setContextValue('roles', [$authenticated->reveal()]);
+    $this->condition->setContextValue('operation', 'INVALID');
+    $this->assertTrue($this->condition->evaluate());
```

Comment:
you can remove the ->assertTrue() call, just execute evaluate() on the condition which will throw the exception already.